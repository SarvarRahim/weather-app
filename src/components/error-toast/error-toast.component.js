import React from 'react';
import { hoc } from '@utils/hoc';
import { useErrorToastProps } from './error-toast.props';
import { SvgIcon } from '@components/atoms';
import ErrorToastImage from '@assets/img/info-icon.svg';
import {
  ErrorToastComponent,
  ErrorToastHolder,
  ErrorToastImg,
  ErrorToastLeft,
  ErrorToastTitle
} from './error-toast.style';

/**
 * <ErrorToast />
 */
export const ErrorToast = hoc(
  useErrorToastProps,
  ({ onCancel, title, children, ...props }) => {
    return (
      <ErrorToastComponent {...props}>
        <ErrorToastHolder>
          <ErrorToastLeft>
            {!children ? (
              <>
                <ErrorToastImg src={ErrorToastImage} />
                <ErrorToastTitle>{title}</ErrorToastTitle>
              </>
            ) : (
              ''
            )}
            {children}
          </ErrorToastLeft>

          <SvgIcon
            iconName='close-icon'
            size='11px'
            $stroke='#fff'
            fill='#fff'
            onClick={onCancel}
          />
        </ErrorToastHolder>
      </ErrorToastComponent>
    );
  }
);
