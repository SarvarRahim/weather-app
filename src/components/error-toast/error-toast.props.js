/**
 * <ErrorToast /> props
 */
export const useErrorToastProps = _ => {};
