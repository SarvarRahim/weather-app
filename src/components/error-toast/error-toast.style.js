import styled from 'styled-components';

export const ErrorToastComponent = styled.div`
  width: 100%;
  height: 100vh;
  position: fixed;
  top: 0;
  left: 0;
  background: transparent;
  overflow: hidden;
  z-index: 22;
  display: ${props => (props.active ? 'block' : 'none')};
`;

export const ErrorToastHolder = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 18px 30px 18px 26px;
  background: #f95050;
  border-radius: 100px;
  position: absolute;
  bottom: 36px;
  left: 0;
  right: 0;
  margin: auto;
  width: 448px;

  svg {
    fill: #fff;
    stroke: #fff;
    cursor: pointer;

    path {
      fill: #fff;
    }
  }
`;

export const ErrorToastLeft = styled.div`
  display: flex;
  align-items: center;
`;

export const ErrorToastImg = styled.img`
  width: 20px;
  height: 20px;
  object-fit: cover;
  margin-right: 15px;
`;

export const ErrorToastTitle = styled.h4`
  font-style: normal;
  font-weight: 500;
  font-size: 18px;
  line-height: 24px;
  letter-spacing: -0.02em;
  color: #ffffff;
`;
