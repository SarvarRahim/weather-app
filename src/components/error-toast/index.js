export * from './error-toast.component';
