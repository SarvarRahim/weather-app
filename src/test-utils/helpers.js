import { render } from '@testing-library/react';
import React, { useRef } from 'react';
import { BrowserRouter, Router } from 'react-router-dom';
import { Provider as ReduxProvider } from 'react-redux';
import { ThemeProvider } from 'styled-components';
import { theme } from '@styles/Theme';
import { store } from '@store';
import { Formik } from 'formik';
import { VideoContext } from '@courses/routes/courses-single/course.context';
import { createMemoryHistory } from 'history';

const AllTheProviders = ({ children, formikValues = {} }) => {
  const videoRef = useRef();
  return (
    <BrowserRouter>
      <ReduxProvider store={store}>
        <ThemeProvider theme={theme}>
          <Formik initialValues={formikValues}>
            <VideoContext.Provider value={videoRef}>
              {children}
            </VideoContext.Provider>
          </Formik>
        </ThemeProvider>
      </ReduxProvider>
    </BrowserRouter>
  );
};

export const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export const renderWithRouter = (
  ui,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) => {
  return {
    ...render(<Router history={history}>{ui}</Router>),
    history
  };
};
