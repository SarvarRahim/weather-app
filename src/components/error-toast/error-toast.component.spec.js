import { customRender } from '@test-utils/helpers';
import { screen } from '@testing-library/react';
import { ErrorToast } from './error-toast.component';

describe('Atoms Context', () => {
  test('Should render ErrorToast component', async () => {
    const { asFragment } = customRender(
      <ErrorToast data-testid='error-toast-component' />
    );
    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByTestId('error-toast-component')).toBeInTheDocument();
  });
});
