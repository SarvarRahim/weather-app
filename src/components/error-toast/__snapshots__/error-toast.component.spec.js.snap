// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms Context Should render ErrorToast component 1`] = `
<DocumentFragment>
  .c5 {
  width: 11px;
  height: 11px;
  fill: #000000;
  stroke: #000000;
  -webkit-transition-duration: 200ms;
  transition-duration: 200ms;
}

.c5[disabled] {
  opacity: 0.2;
}

.c0 {
  width: 100%;
  height: 100vh;
  position: fixed;
  top: 0;
  left: 0;
  background: transparent;
  overflow: hidden;
  z-index: 22;
  display: none;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 18px 30px 18px 26px;
  background: #f95050;
  border-radius: 100px;
  position: absolute;
  bottom: 36px;
  left: 0;
  right: 0;
  margin: auto;
  width: 448px;
}

.c1 svg {
  fill: #fff;
  stroke: #fff;
  cursor: pointer;
}

.c1 svg path {
  fill: #fff;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  width: 20px;
  height: 20px;
  object-fit: cover;
  margin-right: 15px;
}

.c4 {
  font-style: normal;
  font-weight: 500;
  font-size: 18px;
  line-height: 24px;
  -webkit-letter-spacing: -0.02em;
  -moz-letter-spacing: -0.02em;
  -ms-letter-spacing: -0.02em;
  letter-spacing: -0.02em;
  color: #ffffff;
}

<div
    class="c0"
    data-testid="error-toast-component"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <img
          class="c3"
          src="info-icon.svg"
        />
        <h4
          class="c4"
        />
      </div>
      <svg
        class="c5"
        fill="#fff"
      >
        <use
          xlink:href="sprite.svg#close-icon"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;
